packages/containers/src/ComponentForm/kit/flatten.js

```diff
@@ -13,42 +14,50 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
+
+function step(anything, key, payload) {
+	// on native values, we just add it in the payload
+	if (typeof anything !== 'object') {
+		payload[key] = anything;
+	} else {
+		// for objects (including arrays), we flatten it. The we store it in 2 forms.
+		// Let's say that we have a key 'my-object'.
+		// - Form 1: { 'my-object' : {<flattenedObject>} }
+		// - Form 2: { 'my-object.key1': value1, my-object.key2: value2, ... }
+		const subPayload = {};
+		if (Array.isArray(anything)) {
+			anything.forEach((item, index) => {
+				const itemKey = `[${index}]`;
+				step(item, itemKey, subPayload, false);
```

Comment:
the last argument doesn t exists in the function definition, please remove

Reply:
Oh right, I forgot to remove it after the Nth solution :D